File: modules/angular2/test/di/key_spec.ts
import {describe, iit, it, expect, beforeEach} from 'angular2/test_lib';
import {Key, KeyRegistry} from 'angular2/di';

export function main() {
  describe("key", function() {
    var registry: KeyRegistry;

    beforeEach(function() { registry = new KeyRegistry(); });

    it('should be equal to another key if type is the same',
       function() { expect(registry.get('car')).toBe(registry.get('car')); });

    it('should not be equal to another key if types are different',
       function() { expect(registry.get('car')).not.toBe(registry.get('porsche')); });

    it('should return the passed in key',
       function() { expect(registry.get(registry.get('car'))).toBe(registry.get('car')); });

  });
}

File: test/core/di/key_spec.dart
library angular2.test.core.di.key_spec;

import "package:angular2/testing_internal.dart"
    show describe, iit, it, expect, beforeEach;
import "package:angular2/src/core/di/key.dart" show Key, KeyRegistry;

main() {
  describe("key", () {
    KeyRegistry registry;
    beforeEach(() {
      registry = new KeyRegistry();
    });
    it("should be equal to another key if type is the same", () {
      expect(registry.get("car")).toBe(registry.get("car"));
    });
    it("should not be equal to another key if types are different", () {
      expect(registry.get("car")).not.toBe(registry.get("porsche"));
    });
    it("should return the passed in key", () {
      expect(registry.get(registry.get("car"))).toBe(registry.get("car"));
    });
  });
}
